core/sqf/src/seatrans/hbase-trx/src/main/java/org/apache/hadoop/hbase/coprocessor/transactional/SsccRegionEndpoint.java.tmpl

```diff
@@ -2594,7 +2994,76 @@ CoprocessorService, Coprocessor {
     SsccCheckAndPutRegionTxResponse checkAndPutRegionTxResponse = checkAndPutRegionTxResponseBuilder.build();
 
     done.run(checkAndPutRegionTxResponse);
-    */
+  }
+
+  /**
+   * Processes a checkAndDelete operation using a region transaction
+   * @param long transactionId
+   * @param byte[] row
+   * @param byte[] family
+   * @param byte[] qualifier
+   * @param byte[] value
+   * @param Delete delete
+   * @param boolean autoCommit
+   * @return boolean
+   * @throws IOException
+   */
+  public boolean checkAndDeleteRegionTx(long tid, long startId,
+                                byte[] row, byte[] family,
+                                byte[] qualifier, byte[] value,
+                                Delete delete, boolean autoCommit) throws IOException {
+
+    if (LOG.isTraceEnabled()) LOG.trace("Enter checkAndDeleteRegionTx, tid: " + tid);
+    SsccTransactionState state = this.beginTransIfNotExist(tid, startId, true); // This is a regionTX
+    boolean result = false;
+    byte[] rsValue = null;
+
+    Get get = new Get(row);
+    get.addColumn(family, qualifier);
+
+    Result rs = this.get(tid, startId, get);
+
+    boolean valueIsNull = value == null ||
```

Comment:
I assume this is an HBase-ism. I wonder how zero-length varchar strings are represented?